salt/modules/cipher.py

```diff
@@ -0,0 +1,529 @@
+        keyiv += tmp[-1]
+        key = keyiv[:klen]
+        iv = keyiv[klen:klen+ilen]
+    return key, iv
+
+
+def aes_openssl_enc(data, **kwargs):
+    '''
+    Use PyCrypto to encrypt data in openssl aes-256-cbc compatible format.
+
+    If the `key` is larger than 32 bytes.
+    The md5sum of the key will be used as the password.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.aes_openssl_enc datatoenc
+        salt-call cipher.aes_openssl_enc datatoenc key=nevertell
+        salt-call cipher.aes_openssl_enc datatoenc keyfile=/root/.strkey
+    '''
+    # https://github.com/ryanlim/totp-manager/blob/master/totp.py
+    if REQ_ERROR['crypto']:
+        raise Exception(REQ_ERROR['crypto'])
+    key = _get_key(size_limit=32, trim_newline=True, **kwargs)
+    salt = os.urandom(8)
+    key, iv = _openssl_get_key_and_iv(key, salt)
+    if key is None:
+        return None
+    data_pad = 16 - len(data) % 16
+    data = data + (data_pad * chr(data_pad))
+    cypher = AES.new(key, AES.MODE_CBC, iv)
+    data = 'Salted__' + salt + cypher.encrypt(data)
+    return base64.b64encode(data)
+
+
+def aes_openssl_dec(data, **kwargs):
+    '''
+    Use PyCrypto to decrypt data in openssl aes-256-cbc format.
+
+    If the `key` is larger than 32 bytes.
+    The md5sum of the key will be used as the password.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.aes_openssl_dec U2FsdGVkX1+uNDhmDwGjiVk3wpt3uQ=
+        salt-call cipher.aes_openssl_dec U2FsdGVkX1+uNDhmDwGjiVk3wpt3uQ= keyfile=/root/.strkey
+        salt-call cipher.aes_openssl_dec KZQFJEAoUQBIAN52p795SVB7dmfr6iOxsyNpxXX57Gs= key=nevertell
+    '''
+    if REQ_ERROR['crypto']:
+        raise Exception(REQ_ERROR['crypto'])
+    key = _get_key(size_limit=32, trim_newline=True, **kwargs)
+    data = base64.b64decode(data)
+    assert data[:8] == 'Salted__'
+    salt = data[8:16]
+    key, iv = _openssl_get_key_and_iv(key, salt)
+    if key is None:
+        return None
+    ciphertext = data[16:]
+    cipher = AES.new(key, AES.MODE_CBC, iv)
+    padded_plaintext = cipher.decrypt(ciphertext)
+    padding_len = ord(padded_plaintext[-1])
+    plaintext = padded_plaintext[:-padding_len]
+    return plaintext
+
+
+def openssl_enc(data, **kwargs):
+    '''
+    Use openssl to encrypt AES data.
+    Prefer `cipher.aes_enc`
+
+    If the `key` is larger than 32 bytes.
+    The md5sum of the key will be used as the password.
+    If the key is not of block size 16 use PKCS#5 padding
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.openssl_enc datatoenc
+        salt-call cipher.openssl_enc datatoenc key=nevertell
+        salt-call cipher.openssl_enc datatoenc keyfile=/root/.strkey
+    '''
+    if REQ_ERROR['openssl']:
+        raise Exception(REQ_ERROR['openssl'])
+    key = _get_key(size_limit=32, trim_newline=True, **kwargs)
+    # TODO: is this safe... can people see the raw call in ps?
```

Comment:
Yes, this is unsafe. a good way to do this is to pass the stdin manually by using salt vt in salt.utils.vt instead of using a shell